resolver.go

```diff
@@ -151,22 +158,36 @@ func setCommonFlags(msg *dns.Msg) {
 	msg.RecursionAvailable = true
 }
 
+func shuffleAddr(addr []net.IP) []net.IP {
```

Comment:
shuffling is important. The most common API on the resolver side, getaddinfo() doesn't do any shuffling and sends the results as is to the client. In my tests gethostbyname() doesn't do any shuffling either. AFAIK, browsers also typically use the first result and switch to others only if the first one fails.

btw, loadbalancing based on DNS-RR is considered a poor man's solution, irrespective of the order of A records from the DNS server.